### PR TITLE
436 record UUID

### PR DESCRIPTION
**Pull request by Elforama**
Issue: https://github.com/Greenstand/treetracker-android/issues/436

### CODE REVIEW

File: app/src/main/java/org/greenstand/android/TreeTracker/database/entity/PlanterInfoEntity.kt [hunk 1]
```diff
@@ -27,7 +27,9 @@ data class PlanterInfoEntity(
     @ColumnInfo(name = CREATED_AT)
     var createdAt: Long,
     @ColumnInfo(name = BUNDLE_ID)
-    var bundleId: String? = null
+    var bundleId: String? = null,
+    @ColumnInfo(name = RECORD_UUID)
+    var recordUuid: String? = null
```
Comment (arunbakt) [inline app/src/main/java/org/greenstand/android/TreeTracker/database/entity/PlanterInfoEntity.kt:32]: When would the `recordUuid` will be null?  Won't it be a non-null value always?

File: app/src/main/java/org/greenstand/android/TreeTracker/database/Migrations.kt [hunk 2]
```diff
@@ -55,4 +56,10 @@ val MIGRATION_1_2 = object : Migration(1, 2) {
         }
     }
 
+}
+
+val MIGRATION_2_3 = object : Migration(2, 3) {
+    override fun migrate(database: SupportSQLiteDatabase) {
+        database.execSQL("ALTER TABLE ${PlanterInfoEntity.TABLE} ADD COLUMN ${PlanterInfoEntity.RECORD_UUID} TEXT")
```
Comment (arunbakt) [inline app/src/main/java/org/greenstand/android/TreeTracker/database/Migrations.kt:63]: Shouln't we require a non-null value for `record_uuid`?

File: app/src/main/java/org/greenstand/android/TreeTracker/fragments/MapsFragment.kt [hunk 3]
```diff
@@ -236,6 +235,9 @@ class MapsFragment : androidx.fragment.app.Fragment(), OnClickListener, OnMapRea
     }
 
     private fun renderTrees() {
+
+        map ?: return
```
Comment (arunbakt) [inline app/src/main/java/org/greenstand/android/TreeTracker/fragments/MapsFragment.kt:239]: In what conditions would the map be null? When location is not enabled?

Comment (Elforama) [reply app/src/main/java/org/greenstand/android/TreeTracker/fragments/MapsFragment.kt:239]: While working on this it crashed here due to it being null. This is a quick fix

File: app/src/main/java/org/greenstand/android/TreeTracker/database/Migrations.kt (hunk 2 above)
Comment (Elforama) [reply app/src/main/java/org/greenstand/android/TreeTracker/database/Migrations.kt]: Update

File: app/src/main/java/org/greenstand/android/TreeTracker/database/entity/PlanterInfoEntity.kt (hunk 1 above)
Comment (Elforama) [reply app/src/main/java/org/greenstand/android/TreeTracker/database/entity/PlanterInfoEntity.kt]: Updated to non null

**Pull request merged**